session_rpcserver.go

```diff
@@ -317,12 +319,21 @@ func (s *sessionRpcServer) AddSession(_ context.Context,
 
 	sess, err := session.NewSession(
 		req.Label, typ, expiry, req.MailboxServerAddr, req.DevServer,
-		uniquePermissions, caveats, nil, false,
+		uniquePermissions, caveats, nil, false, nil,
 	)
 	if err != nil {
 		return nil, fmt.Errorf("error creating new session: %v", err)
 	}
 
+	// Ensure that the session ID to group ID index is updated with the
+	// new pair before storing the session.
+	err = s.cfg.sessionIDIndex.AddSessionAndGroupIDPair(
```

Comment:
Hmm what if this succeeds when we have multiple linked sessions, and we successfully store the `session_id` for the `group_id` fin the index db, and then `s.db.StoreSession(sess)` below fails.
We'll then have a stored `session_id` in the index for the group that no longer exists, and also the latest sequence number in the index will be pointing to a non existing session.
Currently, this isn't an issue as I see that `ListGroupActions` won't fail due to this and the `GetSessionIDs` function isn't used outside of tests. But I see a risk that we will expand the use of the `GetSessionIDs` function and that people will expect that the results always points to existing sessions, or that we add a `GetLatestSessionIdForGroup` function to the index db that gets the `session_id` for the highest `sequence`, which then might be a non existing session.

So if we want to keep it as is, I think we should at least add comments to the indexes and the `GetSessionIDs` that there's no guarantee that the stored `session_ids` in the index maps to existing stored sessions.

Reply:
Got it! Yeah agree that sqlite would make this so much easier! We should really migrate :)

Reply:
Ok cool - this should no longer be an issue with the latest update :)